src/main/groovy/edu/wpi/first/gradlerio/wpi/dependencies/WPINativeJsonDepRules.groovy

```diff
@@ -27,62 +28,98 @@ class WPINativeJsonDepRules extends RuleSource {
-                String mavenbase = "${cpp.groupId}:${cpp.artifactId}:${WPIVendorDepsExtension.getVersion(cpp.version, wpi)}"
-                String config = cpp.configuration ?: "native_${dep.uuid}"
-                List<String> allPlatforms = platformContainer.collect { Platform p -> p.name }
+                ['debug', ''].each { String buildKind ->
+                    String buildType    = buildKind.contains('debug') ? 'debug' : 'release'
+                    String libSuffix    = buildKind.contains('debug') ? 'd' : ''
+                    String linkSuff     = cpp.sharedLibrary ? '' : 'static'
+                    String name = dep.uuid + cpp.libName
+                    String mavenbase = "${cpp.groupId}:${cpp.artifactId}:${WPIVendorDepsExtension.getVersion(cpp.version, wpi)}"
+                    String config = cpp.configuration ?: "native_${dep.uuid}${buildKind}"
+                    List<String> allPlatforms = platformContainer.collect { Platform p -> p.name }
 
-                // Note: because of a discrepancy between the target platforms of the headers, sources
-                // and the binaries, we don't provide a CombinedNativeLib, meaning all of _binaries,
-                // _headers and _sources must be applied. We do this in WPIVendorDepsExtension#cppVendorLibForBin
+                    // Note: because of a discrepancy between the target platforms of the headers, sources
+                    // and the binaries, we don't provide a CombinedNativeLib, meaning all of _binaries,
+                    // _headers and _sources must be applied. We do this in WPIVendorDepsExtension#cppVendorLibForBin
 
-                if (cpp.headerClassifier != null) {
-                    libs.create("${name}_headers".toString(), NativeLib, { NativeLib lib ->
-                        common(lib)
-                        // Headers apply to all platforms, even if the binaries are missing.
-                        lib.targetPlatforms = allPlatforms
-                        lib.headerDirs << ''
-                        lib.maven = "$mavenbase:${cpp.headerClassifier}@zip"
-                        lib.configuration = config
-                    } as Action<NativeLib>)
-                }
-
-                if (cpp.sourcesClassifier != null) {
-                    libs.create("${name}_sources".toString(), NativeLib, { NativeLib lib ->
-                        common(lib)
-                        // Sources apply to all platforms, even if the binaries are missing.
-                        lib.targetPlatforms = allPlatforms
-                        lib.sourceDirs << ''
-                        lib.maven = "$mavenbase:${cpp.sourcesClassifier}@zip"
-                        lib.configuration = config
-                    } as Action<NativeLib>)
-                }
+                    if (cpp.headerClassifier != null) {
+                        libs.create("${name}_headers${buildKind}".toString(), NativeLib, { NativeLib lib ->

```

Comment:
I think we can take the header and source artifacts outside of the `['debug', '']` loop since they're identical for all artifacts.

Reply:
Fixed